File: MascotasUN/src/main/java/baseDatos/Deserializador.java
package baseDatos;

//Gestion personas
import gestorAplicacion.gestionPersonas.CuentaBancaria;
import gestorAplicacion.gestionPersonas.Proveedor;
import gestorAplicacion.gestionPersonas.Usuario;
import gestorAplicacion.gestionPersonas.Vendedor;
import gestorAplicacion.gestionVentas.*;
//Gestion ventas

import java.io.*;
import java.util.ArrayList;

public class Deserializador {
    /* Se encarga de deserializar las listas de cada clase*/
    public static void deserializar(){
        FileInputStream fileIn;
        //Deserializacion proveedores
        try {
            fileIn = new FileInputStream("src\\baseDatos\\temp\\proveedores.txt");
            ObjectInputStream in = new ObjectInputStream(fileIn);
            ArrayList<Proveedor> proveedores;
            proveedores = (ArrayList<Proveedor>) in.readObject();
            Proveedor.setProveedores(proveedores);
            in.close();
            fileIn.close();
        } catch (FileNotFoundException e){
            e.printStackTrace();
        } catch (IOException e){
            e.printStackTrace();
        } catch (ClassNotFoundException e){
            e.printStackTrace();
        }
        //Deserializacion usuarios
        try {
            fileIn = new FileInputStream("src\\baseDatos\\temp\\usuarios.txt");
            ObjectInputStream in = new ObjectInputStream(fileIn);
            ArrayList<Usuario> usuarios;
            usuarios = (ArrayList<Usuario>) in.readObject();
            Usuario.setUsuarios(usuarios);
            in.close();
            fileIn.close();
        } catch (FileNotFoundException e){
            e.printStackTrace();
        } catch (IOException e){
            e.printStackTrace();
        } catch (ClassNotFoundException e){
            e.printStackTrace();
        }
        //Deserializacion vendedores
        try {
            fileIn = new FileInputStream("src\\baseDatos\\temp\\vendedores.txt");
            ObjectInputStream in = new ObjectInputStream(fileIn);
            ArrayList<Vendedor> vendedores;
            vendedores = (ArrayList<Vendedor>) in.readObject();
            Vendedor.setVendedores(vendedores);
            in.close();
            fileIn.close();
        } catch (FileNotFoundException e){
            e.printStackTrace();
        } catch (IOException e){
            e.printStackTrace();
        } catch (ClassNotFoundException e){
            e.printStackTrace();
        }
        //Deserializacion Cuentas Bancarias
        try {
            fileIn = new FileInputStream("src\\baseDatos\\temp\\cuentasBancarias.txt");
            ObjectInputStream in = new ObjectInputStream(fileIn);
            ArrayList<CuentaBancaria> cuentasBancarias;
            cuentasBancarias = (ArrayList<CuentaBancaria>) in.readObject();
            CuentaBancaria.setCuentasBancarias(cuentasBancarias);
            in.close();
            fileIn.close();
        } catch (FileNotFoundException e){
            e.printStackTrace();
        } catch (IOException e){
            e.printStackTrace();
        } catch (ClassNotFoundException e){
            e.printStackTrace();
        }
        //Deserializacion envios
        try {
            fileIn = new FileInputStream("src\\baseDatos\\temp\\envios.txt");
            ObjectInputStream in = new ObjectInputStream(fileIn);
            ArrayList<Envio> envios;
            envios = (ArrayList<Envio>) in.readObject();
            Envio.setEnvios(envios);
            in.close();
            fileIn.close();
        } catch (FileNotFoundException e){
            e.printStackTrace();
        } catch (IOException e){
            e.printStackTrace();
        } catch (ClassNotFoundException e){
            e.printStackTrace();
        }
        //Deserializacion facturas
        try {
            fileIn = new FileInputStream("src\\baseDatos\\temp\\facturas.txt");
            ObjectInputStream in = new ObjectInputStream(fileIn);
            ArrayList<Factura> facturas;
            facturas = (ArrayList<Factura>) in.readObject();
            Factura.setFacturas(facturas);
            in.close();
            fileIn.close();
        } catch (FileNotFoundException e){
            e.printStackTrace();
        } catch (IOException e){
            e.printStackTrace();
        } catch (ClassNotFoundException e){
            e.printStackTrace();
        }
      //Deserializacion gatos
        try {
            fileIn = new FileInputStream("src\\baseDatos\\temp\\gatos.txt");
            ObjectInputStream in = new ObjectInputStream(fileIn);
            ArrayList<Gato> gatos;
            gatos = (ArrayList<Gato>) in.readObject();
            Gato.setGatos(gatos);
            in.close();
            fileIn.close();
        } catch (FileNotFoundException e){
            e.printStackTrace();
        } catch (IOException e){
            e.printStackTrace();
        } catch (ClassNotFoundException e){
            e.printStackTrace();
        }
        //Deserializacion perros
        try {
            fileIn = new FileInputStream("src\\baseDatos\\temp\\perros.txt");
            ObjectInputStream in = new ObjectInputStream(fileIn);
            ArrayList<Perro> perros;
            perros = (ArrayList<Perro>) in.readObject();
            Perro.setPerros(perros);
            in.close();
            fileIn.close();
        } catch (FileNotFoundException e){
            e.printStackTrace();
        } catch (IOException e){
            e.printStackTrace();
        } catch (ClassNotFoundException e){
            e.printStackTrace();
        }
        //Deserializacion productos
        try {
            fileIn = new FileInputStream("src\\baseDatos\\temp\\productos.txt");
            ObjectInputStream in = new ObjectInputStream(fileIn);
            ArrayList<Producto> productos;
            productos = (ArrayList<Producto>) in.readObject();
            Producto.setProductos(productos);
            in.close();
            fileIn.close();
        } catch (FileNotFoundException e){
            e.printStackTrace();
        } catch (IOException e){
            e.printStackTrace();
        } catch (ClassNotFoundException e){
            e.printStackTrace();
        }
        //Deserializar tienda
        try {
            fileIn = new FileInputStream("src\\baseDatos\\temp\\tienda.txt");
            ObjectInputStream in = new ObjectInputStream(fileIn);
            ArrayList<Tienda> tienda;
            tienda = (ArrayList<Tienda>) in.readObject();
            Tienda.setTienda(tienda);
            in.close();
            fileIn.close();
        } catch (FileNotFoundException e){
            e.printStackTrace();
        } catch (IOException e){
            e.printStackTrace();
        } catch (ClassNotFoundException e){
            e.printStackTrace();
        }
    }
}

File: Practica2-MascotasUN/src/baseDatos/Deserializador.py
import pickle
import os
import pathlib
from src.gestorAplicacion.gestionPersonas.Gato import Gato
from src.gestorAplicacion.gestionPersonas.Gato import Perro
from src.gestorAplicacion.gestionVentas.Producto import Producto
from src.gestorAplicacion.gestionPersonas.Vendedor import Vendedor
from src.gestorAplicacion.gestionPersonas.Proveedor import Proveedor
from src.gestorAplicacion.gestionPersonas.Vendedor import Usuario
from src.gestorAplicacion.gestionVentas.Envio import Envio
from src.gestorAplicacion.gestionVentas.Factura import Factura
from src.gestorAplicacion.gestionVentas.Tienda import Tienda
from src.gestorAplicacion.gestionPersonas.CuentaBancaria import CuentaBancaria
class Deserializador:
    @classmethod
    def deserializar(cls):
        baseDatos = {
            "gato":lambda x :Gato.setGatos(x),
            "perro":lambda x :Perro.setPerros(x),
            "producto":lambda x :Producto.setProductos(x),
            "vendedor":lambda x :Vendedor.setVendedores(x),
            "proveedor":lambda x :Proveedor.setProveedores(x),
            "usuario":lambda x :Usuario.setUsuarios(x),
            "factura":lambda x :Factura.setFacturas(x),
            "tienda":lambda x :Tienda.setTienda(x),
            "cuentaBancaria":lambda x :CuentaBancaria.setCuentasBancarias(x),
            "inventario":lambda x :Tienda.setInventario(x),
            "catalogo":lambda x :Tienda.setCatalogo(x)
        }

        for arch, set in baseDatos.items():
            picklefile = open(os.path.join(pathlib.Path(__file__).parent.absolute(), f"tmp\\{arch}"), "rb")
            dato = pickle.load(picklefile)
            set(dato)
            picklefile.close()